apps/portal/js/ues-dashboards.js

```diff
@@ -109,46 +141,75 @@
      */
     var convertToDesignerLayout = function(json) {
     
-        var componentBoxListHbs = Handlebars.compile($("#ues-component-box-list-hbs").html() || ''), 
+        var componentBoxListHbs = Handlebars.compile($("#ues-component-box-list-hbs").html() || ''),
             container = $('<ul />');
         
-        $(componentBoxListHbs(json)).appendTo(container).html();
+        $(componentBoxListHbs(json)).appendTo(container)
         
         return container;
     }
-
+    
+    /**
+     * renders a page in the dashboard designer and the view modes
+     * @param element
+     * @param dashboard
+     * @param page
+     * @param pageType
+     * @param done
+     * @param isDesigner
+    */
     var renderPage = function (element, dashboard, page, pageType, done, isDesigner) {
+        
         setDocumentTitle(dashboard, page);
         wirings = wires(page, pageType);
-        var container;
-        var area;
         
-        var layout = (pageType === 'anon' ?  $(page.layout.content.anon) : $(page.layout.content.loggedIn));
-        var content = page.content[pageType];
+        var layout = (pageType === 'anon' ?  $(page.layout.content.anon) : $(page.layout.content.loggedIn)),
+            content = page.content[pageType], 
+            componentBoxContentHbs = Handlebars.compile($('#ues-component-box-content-hbs').html() || '');
         
         // this is to be rendered only in the designer. in the view mode, the template is rendered in the server side.
         if (isDesigner) { 
             element.html(convertToDesignerLayout(layout[0]));
         }
         
-        for (area in content) {
-            if (content.hasOwnProperty(area)) {          
-                container = $('#' + area, element);
-                content[area].forEach(function (options) {
-                    createComponent(container, options, function (err) {
-                        if (err) {
-                            console.error(err);
-                        }
-                    });
+        $('.ues-component-box').each(function (i, container) {
+            container = $(container);
+            
+            var id = container.attr('id'),
+                hasComponent = content.hasOwnProperty(id) && content[id][0];
+            
+            // the component box content (the skeleton) should be added only in the designer mode and 
+            // the view mode only if there is a component
+            if (isDesigner || hasComponent) {
+                container.html(componentBoxContentHbs());    
+            }
+            
+            if (container.attr('data-banner') == 'true') {
+                container.find('.ues-component-body').addClass('ues-banner-placeholder');
+            }
+            
+            if (hasComponent) {
+                createComponent(container, content[id][0], function (err) {
+                    if (err) {
+                        console.error(err);
```

Comment:
Removed the console.error. Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174